nifi-api/src/main/java/org/apache/nifi/processor/AbstractProcessor.java

```diff
@@ -27,7 +30,12 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
             onTrigger(context, session);
             session.commit();
         } catch (final Throwable t) {
-            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            StringWriter stacktraceWriter = new StringWriter();
```

Comment:
that is actually the current behavior/intent.  I think the change needed is to simply get rid of the 'if debug enabled' stuff we did in the logger and instead just always log the stacktrace when it is there.  This is what Adam Taft had advocated for previously as I recall.  I now see why he was saying it.  So yeah in the UI it should be a short and ideally meaningful message (not always possible) and if a throwable shows up we should put the whole stack trace in the logs.

The current idea that a user will go in an do debug enabled....just has proven to be a failed experiment in my view and as someone who advocated for that I now think i was wrong.